src/lib/Components/Inbox/Conversations/Preview/InvoicePreview.tsx

```diff
@@ -99,29 +111,77 @@ const InvoiceStateButton: React.SFC<InvoiceStateButtonProps> = ({ state }) => {
   }
 }
 
-export const InvoicePreview: React.SFC<Props> = ({ invoice, onSelected }) =>
-  <TouchableHighlight onPress={onSelected} underlayColor={colors["gray-light"]}>
-    <Container>
-      <Icon source={require("../../../../../../images/payment_request.png")} />
-      <TextContainer>
-        <PaymentRequest>Payment request</PaymentRequest>
-        <CostLabel>
-          {invoice.total}
-        </CostLabel>
-      </TextContainer>
-      <TextContainer>
-        <InvoiceStateButton state={invoice.state} />
-      </TextContainer>
-    </Container>
-  </TouchableHighlight>
-
-export default createFragmentContainer(
+interface State {
+  hasReceivedNotification: boolean
+  subscription: EmitterSubscription
```

Comment:
In case you haven’t read about [property declarations](https://www.typescriptlang.org/docs/handbook/classes.html) yet, it should probably look like:

```ts
class InvoicePreview {
  private readonly subscription: EmitterSubscription

  constructor(props) {
    super(props)
    this.subscription = notificationsManagerEmitter.addListener("PaymentRequestPaid", paymentRequestPaidCallback)
  }

  componentWillUnmount() {
    this.subscription.remove()
  }
}
```